### ISSUE TITLE
After updating KanjiDIC2 self test fails

### ISSUE THREAD

**Issue by begal** [mvysny/aedict#567]
Hello.

Aedict is crashing in my tablet after updating dictionaries. Uninstalling and reinstalling makes no difference.

I took my phone, run the self test and it was 100% ok, then updated dictionaries and it updated KanjiDIC2, run self test again and 5 tests crashed:
ResolveKanjis本
ResolveKanjis当
ResolveKanjis本当
KanjidicQuery
SkipQuery

**Comment by seizonsenryaku** [comment]
I have the same issue...

**Comment by mvysny** [comment]
I apologize for this behavior. I have published an updated kanjidic2 and my backward compatibility routines failed me. Fixing now, I will publish fixed Aedict 3.36 today.
Probable cause:
```
java.lang.RuntimeException: Loader failed: java.lang.RuntimeException: Failed to run query kanji:農: kanji:"農"
at sk.baka.aedict3.util.android.list.CheckableListFragment.handleLoaderError(CheckableListFragment.java:94)
at sk.baka.aedict3.dict.AbstractEntryRefListFragment.handleDictionaryCrash(AbstractEntryRefListFragment.java:244)
at sk.baka.aedict3.dict.KanjiDicListFragment.handleLoaderError(KanjiDicListFragment.java:246)
at sk.baka.aedict3.util.android.list.CheckableListFragment$1.onFailure(CheckableListFragment.java:113)
at sk.baka.aedict3.util.android.LoaderEx$Loader.onFailure(LoaderEx.java:181)
at sk.baka.aedict3.util.android.LoaderEx$Loader.access$100(LoaderEx.java:105)
at sk.baka.aedict3.util.android.LoaderEx$Loader$1.run(LoaderEx.java:158)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:211)
at android.app.ActivityThread.main(ActivityThread.java:5317)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1016)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
Caused by: java.lang.RuntimeException: Failed to run query kanji:農: kanji:"農"
at sk.baka.aedict.search.lucene.LuceneSearchUtils.search(LuceneSearchUtils.java:153)
at sk.baka.aedict.search.impl.LuceneCache$1.search(LuceneCache.java:87)
at sk.baka.aedict.search.KanjidicQuery$ResolveKanjis.find(KanjidicQuery.java:220)
at sk.baka.aedict.search.KanjidicQuery$ResolveKanjis.search(KanjidicQuery.java:190)
at sk.baka.aedict3.dict.KanjiDicListFragment$MyLoader.loadItemsInBackground(KanjiDicListFragment.java:53)
at sk.baka.aedict3.util.android.LoaderEx$Loader.run(LoaderEx.java:129)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: Failed to invoke sk.baka.aedict.dict.Kanjidic2Entry.readFrom(DataInput): java.lang.ArrayIndexOutOfBoundsException: length=19; index=19
at sk.baka.aedict.util.Writables.read(Writables.java:247)
at sk.baka.aedict.search.lucene.LuceneSearchUtils.toWritables(LuceneSearchUtils.java:124)
at sk.baka.aedict.search.lucene.LuceneSearchUtils.search(LuceneSearchUtils.java:143)
... 10 more
Caused by: java.lang.RuntimeException: Failed to invoke sk.baka.aedict.dict.Kanjidic2Entry.readFrom(DataInput): java.lang.ArrayIndexOutOfBoundsException: length=19; index=19
at sk.baka.aedict.util.Writables.read(Writables.java:227)
at sk.baka.aedict.util.Writables.read(Writables.java:245)
... 12 more
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at sk.baka.aedict.util.Writables.read(Writables.java:221)
... 13 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=19; index=19
at sk.baka.aedict.util.Box$Value.enumMapOfBasics(Box.java:1111)
at sk.baka.aedict.dict.Kanjidic2Entry.unbox(Kanjidic2Entry.java:217)
at sk.baka.aedict.dict.Kanjidic2Entry.readFrom(Kanjidic2Entry.java:172)
... 16 more
```

**Comment by mvysny** [comment]
Fixed in Aedict 3.36
